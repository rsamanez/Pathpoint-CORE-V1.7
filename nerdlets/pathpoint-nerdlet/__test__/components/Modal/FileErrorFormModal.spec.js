import React from 'react';
import { create } from 'react-test-renderer';
import {
  BodyFileErrorFormModal,
  HeaderFileErrorFormModal
} from '../../../components/Modal/FileErrorFormModal';

describe('FileErrorFormModal component', () => {
  test('Render body with default data', () => {
    const body = create(
      <BodyFileErrorFormModal
        SetConfigurationJSON={jest.fn()}
        validateKpiQuery={{}}
        _onClose={jest.fn()}
        errorsList={[
          { dataPath: '/data/0', message: 'message error 1' },
          { dataPath: '/data/1', message: 'message error 2' }
        ]}
      />
    );
    expect(body.toJSON()).toMatchSnapshot();
  });

  test('Render header with default data', () => {
    const header = create(<HeaderFileErrorFormModal />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});
