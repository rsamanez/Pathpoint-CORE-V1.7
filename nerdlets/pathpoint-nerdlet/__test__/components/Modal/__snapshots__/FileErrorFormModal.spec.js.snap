// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileErrorFormModal component Render body with default data 1`] = `
Array [
  <div
    className="containerError"
  >
    <div
      className="error-alert-modal"
    >
      <p>
        /data/0 message error 1
      </p>
    </div>
    <div
      className="error-alert-modal"
    >
      <p>
        /data/1 message error 2
      </p>
    </div>
  </div>,
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "justifyContent": "flex-end",
      }
    }
  >
    <label
      className="button"
      color="primary"
      htmlFor="file-upload"
    >
      Fix & Upload
    </label>
    <input
      accept=".json"
      id="file-upload"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>,
]
`;

exports[`FileErrorFormModal component Render header with default data 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="titleModal"
  >
    <p
      className="error-modal-title"
    >
      Your view configuration file has some errors, please try again
    </p>
  </div>
</div>
`;
